### ISSUE TITLE
Option to have reverse numbering of identical heading ids

### ISSUE THREAD

**Issue by eugenesvk** [Keats/gutenberg#2176]
Currently identical ids are disambiguated by adding values from top to bottom, so

##### About
↑ `about`
##### About
↑ `about-1`

However, this means that if you keep adding new sections, e.g., `Section 2023`, **every single id** will change, which breaks links and would require manual intervention, which is less than ideal

Instead if would be nice to have an option to generate those numerical ids in reverse order, then the only ids that will change will be for the 1st section that used to be the latest, but becomes 2nd

**Comment by Keats** [comment]
I don't think it would be a good solution since you links are still broken with the proposed solution. The fix for the end user is to hardcode a different anchor when adding a header with a conflicting anchor.

**Comment by eugenesvk** [comment]
But **they**'re not broken, only the **one item** is temporary without a permalink instead of all items currently. And even then, it's not really fully broken -  no-id link always refers to the most current item, which makes sense for this chronological scheme

Though the proper solution there would be to just add an id there in the first item as well, so instead of 
Section 2022
About = `about`

you'd have `about-2`

and then when it moves down, the link doesn't change, it's already the second item and will always be the second one from the bottom
Then the only broken link will be the first of all the dupes ever, not every dupe for a brief period while it's fresh

However, you'd then lose the `about` = always the most recent section. To solve this you'd need to allow two links for the same header

And this is still better vs. having to manually add an id to each section, which you can have many, and then you can have many sections, so you just won't bother, just like a lot of changelogs on github are broken since every new section has the same subsections, and those get ids from top to bottom

**Comment by Keats** [comment]
Sorry I meant broken as not pointing to the same thing, it wasn't the right wording.

I don't really understand how going from bottom to top helps anything. If you add something at the end all the ids will change as well. Appending content is probably even more common than prepending in practice.

You only need to hardcode ids to sections you are adding, which is the compromise you would need to do _either way_ in both cases if you don't want to change existing links.

**Comment by eugenesvk** [comment]
> Sorry I meant broken as not pointing to the same thing, it wasn't the right wording.

It's fine, that's how I understood it. Though as I mentioned, it's only broken for one first-ever item with this scheme. Though, see below, if you know in advance that it's not going to be the only section forever, then nothing will be broken

> I don't really understand how going from bottom to top helps anything.

It helps by automatically preserving all the links when you add repetitive content to the top

>   If you add something at the end all the ids will change as well. Appending content is probably even more common than prepending in practice.

This is a flexible option, not applied to every content category, so:
  - the lists that are append-only will use the current scheme
  - the lists that are prepend-only will use the reverse scheme
  - the lists that do both will have to use manual overrides

> You only need to hardcode ids to sections you are adding, which is the compromise you would need to do _either way_ in both cases if you don't want to change existing links.

Not really, if multiple identical sections all have numerical ids, then I wouldn't have to touch the new sections. I'd only have to touch the very first section if I know that it's going to become duplicate later on

So 
```
# 2021
## A {#A-1}
```
↑ manually add `A-1` so that later when a new section is added this will stay `A-1` even when this overrid

then a new section would remove he manual override from the old section

↓ this will be `A-2`
```
# 2022
## A

# 2021
## A
```
↑ this will become `A-1` automatically and thus retain the link
